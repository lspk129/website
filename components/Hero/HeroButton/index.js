import styled from 'styled-components';

const Button = styled.button`
  position: relative;
  display: block;
  margin: 30px auto;
  overflow: hidden;
  border-width: 0;
  outline: none;
  border-radius: 2px;
  box-shadow: 0 1px 4px rgba(0, 0, 0, .6);
  background-color: #ff6666;
  color: #fff;
  transition: background-color .3s;
  &:hover {
    background-color: #fff;
    color: #ff6666
  };
  padding: 15px 50px;
  top: 65%;
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
`;

export default Button;
