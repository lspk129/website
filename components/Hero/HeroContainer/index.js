import styled from 'styled-components';

const HeroContainer = styled.div`
  padding: 100px 0px;
  justify-self: center;
  text-align: center;
`;

export default HeroContainer;
