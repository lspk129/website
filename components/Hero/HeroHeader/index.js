import styled from 'styled-components';

const HeaderContainer = styled.div`
  grid-row: 1 / 2;
  align-text: center;
  justify-self: center;
`;

export default HeaderContainer;
