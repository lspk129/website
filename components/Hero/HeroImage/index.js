import styled from 'styled-components';

const HomeImage = styled.div`
  background-image: url("/static/Hero_image.jpg");
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  display: grid;
  text-align: center;
`;

export default HomeImage;
