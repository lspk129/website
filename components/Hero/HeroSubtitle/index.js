import React from 'react';
import styled from 'styled-components';

const HeroSubtitle = styled.h2`
    text-align: center;
    position: absolute;
    top: 55%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: white;
`;


export default ({ children }) => (
  <HeroSubtitle>
    { children }
  </HeroSubtitle>
);
