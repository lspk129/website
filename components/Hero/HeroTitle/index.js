import styled from 'styled-components';

const HeroTitle = styled.h1`
  text-align: center;
  position: absolute;
  top: 35%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: white;
  font-size: 4rem;
  white-space: nowrap;
  font-weight: bold;
  text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
  line-height: 4rem;
`;

export default HeroTitle;
