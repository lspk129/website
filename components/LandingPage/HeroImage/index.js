import styled from 'styled-components';

export default styled.div`
  background-image: url("http://trendytheme.net/demo2/wp/69/multipage/wp-content/uploads/revslider/web-product-dark/webproduct_darkbg2.jpg");
  background-position: center center;
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  margin: -10px;
  height: 100vh
`;

