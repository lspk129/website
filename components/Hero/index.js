import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import { Margin, Padding } from 'styled-components-spacing';

import HeroImage from './HeroImage';
import HeroTitle from './HeroTitle';
import Subtitle from '../atoms/Subtitle';

const Hero = () => (
  <HeroImage>
    <Margin all={5}>
      <HeroTitle>
        Moka Haiku:<br />web app agency
      </HeroTitle>
      <Padding bottom={4}>
        {/* <HeroSubtitle>*/}
        <Subtitle white>
          Creating fast & interactive web applications with React and Meteor
        </Subtitle>
        {/* </HeroSubtitle>*/}
      </Padding>
      <RaisedButton label="get a free consultation" secondary />
    </Margin>
  </HeroImage>
);

export default Hero;
