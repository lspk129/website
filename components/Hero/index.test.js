import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RaisedButton from 'material-ui/RaisedButton';
import getMuiTheme from 'material-ui/styles/getMuiTheme';


import Hero from './';
import HeroImage from './HeroImage';
import HeroHeader from './HeroHeader';
import HeroContainer from './HeroContainer';
import HeroTitle from './HeroTitle';
import HeroSubtitle from './HeroSubtitle';

describe('Hero', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, {
    context: { muiTheme },
  });

  it('renders correctly', () => {
    const wrapper = shallowWithContext(<Hero />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders Hero component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
  });

  it('renders HeroHeader component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(HeroHeader)).toBePresent();
  });

  it('renders HeroContainer component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(HeroContainer)).toBePresent();
  });

  it('renders HeroImage component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(HeroImage)).toBePresent();
  });

  it('renders HeroTitle component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(HeroTitle)).toBePresent();
  });

  it('renders HeroSubtitle component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(HeroSubtitle)).toBePresent();
  });

  it('renders RaisedButton component', () => {
    const wrapper = shallowWithContext(<Hero />);
    expect(wrapper).toBePresent();
    expect(wrapper.find(RaisedButton)).toBePresent();
    expect(wrapper.find(RaisedButton)).toHaveProp('label', 'get a free consultation');
  });
});
