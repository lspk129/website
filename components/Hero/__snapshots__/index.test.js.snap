// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<styled.div>
  <Styled(Component)
    all={5}
  >
    <styled.h1>
      Moka Haiku:
      <br />
      web app agency
    </styled.h1>
    <Styled(Component)
      bottom={4}
    >
      <styled.h2
        white={true}
      >
        Creating fast & interactive web applications with React and Meteor
      </styled.h2>
    </Styled(Component)>
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="get a free consultation"
      labelPosition="after"
      primary={false}
      secondary={true}
    />
  </Styled(Component)>
</styled.div>
`;
